### PR TITLE
outlier.fill added to geom_boxplot

### LINKED ISSUE #1785: geom_boxplot fill passed onto outliers
Is it possible to pass the fill value form geom_boxplot aesthetic to the outlier fill color?  This would allow the fill of the outlier points to match the fill of the box plot if the point is set to a shape that allows for a fill.

Line https://github.com/hadley/ggplot2/blob/master/R/geom-boxplot.r#L211 could be something like 

```{r}
# definition
outlier.fill = NULL,
...

# outlier data.frame
fill = outlier.fill %||% data$fill[1]
```

The example would be 

```{r}
ggplot(iris, aes(Species, Sepal.Length, fill = Species)) + geom_boxplot(outlier.shape = 21)
```
screen shot 2016-09-27 at 3 46 25 pm

The legend isn't correct, but the desired output would be something like...
```{r}
ggplot(iris, aes(Species, Sepal.Length)) + geom_boxplot(mapping = aes(color = Species)) + geom_boxplot(outlier.shape = 21, fill = NULL)
```
screen shot 2016-09-27 at 3 45 49 pm

Thank you,
Barret

### PR DESCRIPTION
**Pull request by schloerke**
Fixes #1785

Before (no fill color in points):
screen shot 2016-09-27 at 4 22 15 pm

New output:

```{r}
load_all(); ggplot(iris, aes(Species, Sepal.Width, fill = Species)) + geom_boxplot(outlier.shape = 21)
```
screen shot 2016-09-27 at 4 15 23 pm

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,92 +1,97 @@
-# ggplot2 2.1.0.9000 
+# ggplot2 2.1.0.9000
```
Comment (hadley) [inline NEWS.md]: Could you please fix up these spurious ws changes?

File: NEWS.md [hunk 2]
```diff
@@ -1,92 +1,97 @@
...
 
 * The documentation for theme elements has been improved (#1743).
 
 * `geom_boxplot` gain new `outlier.alpha` argument for controlling the alpha of
-   outlier points independently of the alpha of the boxes. 
+   outlier points independently of the alpha of the boxes.
    Analogous to outlier.colour, outlier.shape, etc.
    (@jonathan-g).
 
+* `geom_boxplot` gain new `outlier.fill` argument for controlling the fill of
```
Comment (hadley) [inline NEWS.md]: Can you put the new news at the top please?

**Comment by schloerke** [comment]
@hadley  I've removed all extra changes!

(current travis failures are not due to this PR's changes as they passed earlier)

**Comment by hadley** [comment]
Thanks!

**Pull request closed without merging**